sdks/python/apache_beam/transforms/async_dofn.py

```diff
@@ -256,10 +256,12 @@ def schedule_item(self, element, ignore_buffer=False, *args, **kwargs):
         total_sleep += sleep_time
         sleep(sleep_time)
 
-  def next_time_to_fire(self):
+  def next_time_to_fire(self, key):
+    random.seed(key)
     return (
         floor((time() + self._timer_frequency) / self._timer_frequency) *
-        self._timer_frequency)
+        self._timer_frequency) + (
+            random.random() * self._timer_frequency)
```

Comment:
I feel like doing all of the work to find a round increment of `_timer_frequency` is wasted compute once you add the extra fuzziness of `random.random() * self._timer_frequency` since you're no longer on a round increment afterwards

Reply:
Ah I see, makes sense. There may be a way to queue up per-key firing times, but that's a more substantial piece of work since Beam timers don't work like that themselves. The compute here is negligible so I'm not particularly worried about it, if we wanted to use some memory instead we could pick an offset per-key and store it, that would eliminate the spikes but keep a consistent cadence of each key firing at the desired frequency when in steady state